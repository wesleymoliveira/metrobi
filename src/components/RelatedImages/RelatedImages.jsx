import { Typography } from '../Typography/Typography';
import * as S from './RelatedImages.styled';

export function RelatedImages() {
	return (
		<S.Wrapper data-testid='related-images'>
			<Typography size='xxlarge'>
				Related images
			</Typography>
		</S.Wrapper>
	);
}
