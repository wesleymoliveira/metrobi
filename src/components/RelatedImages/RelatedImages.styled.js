import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.relatedImagesBackground};
    width: 80%;
    height: 12rem;
    display: flex;
    justify-content: center;
    align-items: start;
  `}
`;
