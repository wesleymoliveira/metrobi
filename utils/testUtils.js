import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
