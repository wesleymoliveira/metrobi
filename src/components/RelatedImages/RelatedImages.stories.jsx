import React from 'react';
import { RelatedImages } from './RelatedImages';

const RelatedImagesArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/RelatedImages',
	component: RelatedImages,
	argTypes: RelatedImagesArgTypes
};

export const Default = (args) => <RelatedImages {...args} />;
