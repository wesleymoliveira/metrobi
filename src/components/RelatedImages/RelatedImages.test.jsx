import { screen } from '@testing-library/react';

import { RelatedImages } from './RelatedImages';
import { render, snapshotRenderer } from '../../../utils/testUtils';

describe('<RelatedImages />', () => {
		it('should render the component', () => {
		render(<RelatedImages />);
		
		expect(screen.getByTestId(/related-images/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<RelatedImages />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original background-color', () => {
		const tree = snapshotRenderer(<RelatedImages />).toJSON();
		//make sure the background-color is the original one
		expect(tree).toHaveStyleRule('background-color', '#2BB673')
	});

});
