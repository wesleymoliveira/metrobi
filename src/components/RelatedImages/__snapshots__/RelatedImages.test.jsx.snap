// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedImages /> should match snapshot 1`] = `
.c1 {
  color: #FAFAFA;
  font-size: 2.8rem;
}

.c0 {
  background-color: #2BB673;
  width: 80%;
  height: 12rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<div
  className="c0"
  data-testid="related-images"
>
  <h1
    className="c1"
    color="white"
    data-testid="typography"
    size="xxlarge"
  >
    Related images
  </h1>
</div>
`;
